### PR TITLE
Fixed driver sharing due to static keyword

### PR DESCRIPTION
**Pull request by ninadbstack**
- Fixed static driver causing platform name mismatch for devices
- Reduced duplication

### CODE REVIEW

File: src/main/java/io/percy/appium/metadata/AndroidMetadata.java [hunk 1]
```diff
@@ -110,28 +107,7 @@ public String getDisplaySysDump() {
         return resultString;
     }
 
-
-    public String orientation() {
```
Comment (ninadbstack) [inline src/main/java/io/percy/appium/metadata/AndroidMetadata.java:114]: Moving duplicated code to base class

File: src/main/java/io/percy/appium/metadata/Metadata.java [hunk 2]
```diff
@@ -1,19 +1,21 @@
 package io.percy.appium.metadata;
 
+import org.openqa.selenium.ScreenOrientation;
+
 import io.appium.java_client.AppiumDriver;
 
 public abstract class Metadata {
-    private static AppiumDriver driver;
```
Comment (ninadbstack) [inline src/main/java/io/percy/appium/metadata/Metadata.java:6]: This should not be static - which is causing the bug

File: src/main/java/io/percy/appium/metadata/AndroidMetadata.java [hunk 3]
```diff
@@ -110,28 +107,7 @@ public String getDisplaySysDump() {
...
-          Object orientationCapability = driver.getCapabilities().getCapability("orientation");
-          if (orientationCapability != null) {
-              return orientationCapability.toString().toLowerCase();
-          } else {
-              return "portrait";
-          }
-      }
-  }
-
+    protected ScreenOrientation driverGetOrientation() {
```
Comment (prklm10) [inline src/main/java/io/percy/appium/metadata/AndroidMetadata.java:110]: `driverGetOrientation` name looks weird. Can we use something like getCurrentOrientation or checkOrientation ?

**Review by prklm10 — APPROVED** [review]

File: src/main/java/io/percy/appium/metadata/AndroidMetadata.java (hunk 3 above)
Comment (ninadbstack) [reply src/main/java/io/percy/appium/metadata/AndroidMetadata.java:110]: We already have orientation function. Which is external ( public ) function. Both names suggested are close to that. 

So rather used driver prefix as it does driver.getOrientation and made it protected. 

Had to do cause appiumdriver does not have get orientation but android and ios driver does

Comment (ninadbstack) [reply src/main/java/io/percy/appium/metadata/AndroidMetadata.java:110]: Open to any other better name though, which doesnt sound an alternative to orientation getter

Comment (prklm10) [reply src/main/java/io/percy/appium/metadata/AndroidMetadata.java:110]: Np. As this is protected customer will not access it. We can go with the above name.

**Pull request merged**